mutiny-core/src/ldkstorage.rs

```diff
@@ -88,26 +92,55 @@ impl<S: MutinyStorage> MutinyNodePersister<S> {
         format!("{}_{}", key, self.node_id)
     }
 
-    fn persist_local_storage<W: Writeable>(
+    fn init_persist_local_storage<W: Writeable>(
         &self,
-        key: &str,
+        key: String,
         object: &W,
-        version: Option<u32>,
-    ) -> Result<(), lightning::io::Error> {
-        let key_with_node = self.get_key(key);
-        self.storage
-            .set_data(key_with_node, object.encode(), version)
-            .map_err(|e| {
-                match e {
-                    MutinyError::PersistenceFailed { source } => {
-                        log_error!(self.logger, "Persistence failed on {key}: {source}");
+        version: u32,
+        update_id: MonitorUpdateIdentifier,
+    ) -> ChannelMonitorUpdateStatus {
+        let storage = self.storage.clone();
+        let chain_monitor = self.chain_monitor.clone();
+        let logger = self.logger.clone();
+        let object = object.encode();
+        spawn(async move {
+            loop {
+                // Sleep before persisting to give chance for the manager to be persisted
+                sleep(50).await;
```

Comment:
Locally 50ms is probably fine, but remotely it would need to be something like 500ms?

Reply:
That'd be really slow

Reply:
it's just not safe otherwise unfortunately. Depends if we want to nip this out of sync issue in the butt or not.

Reply:
Better than users losing funds.

Reply:
>  we only have to delay until its persistence has started

This is a really good point, should cover most scenarios besides maybe a temporary network blip that affected the manager save but not the monitor.